src/acl_source.f90

```diff
@@ -494,22 +497,32 @@ subroutine Compute_Momentum_Source_Term_pointwise
       t1 = MPI_WTIME()
 
       ! Add the source term
-      do k=1,xsize(3)
-         zmesh=real(k+xstart(3)-1-1,mytype)*dz
-         do j=1,xsize(2)
-            if (istret.eq.0) ymesh=real(xstart(2)+j-1-1,mytype)*dy
-            if (istret.ne.0) ymesh=yp(xstart(2)+j-1)
-            do i=1,xsize(1)
-               xmesh=(i-1)*dx
-               do isource=1,NSource
-                  dist = sqrt((Sx(isource)-xmesh)**2+(Sy(isource)-ymesh)**2+(Sz(isource)-zmesh)**2)
-                  epsilon=eps_factor*(dx*dy*dz)**(1.0/3.0)
-                  if (dist<10.0*epsilon) then
-                     Kernel= one/(epsilon**3.0*pi**1.5)*exp(-(dist/epsilon)**2.0)
-                  else
-                     Kernel=zero
-                  endif
+      epsilon = (eps_factor*(dx*dy*dz)**(1.0/3.0)) ! FIXME compatible with stretching ?
+      epsilon2 = epsilon**2
+      dist2_max = 100._mytype * epsilon2
+      prefactor = 1._mytype/(epsilon**3*pi**1.5)
+      invepsilon2 = 1._mytype / epsilon2
+      do isource=1,NSource
+         do k=1,xsize(3)
+            zmesh=real(k+xstart(3)-1-1,mytype)*dz
+            distz2 = (Sz(isource)-zmesh)**2
+            if (distz2 > dist2_max) cycle
+            do j=1,xsize(2)
+               if (istret.eq.0) then
+                  ymesh=real(xstart(2)+j-1-1,mytype)*dy
+               else
+                  ymesh=yp(xstart(2)+j-1)
+               end if
+               disty2 = distz2 + (Sy(isource)-ymesh)**2
+               if (disty2 > dist2_max) cycle
+               do i=1,xsize(1)
+                  xmesh=(i-1)*dx
+                  dist2 = disty2 + (Sx(isource)-xmesh)**2
+                  !if (dist2 > dist2_max) cycle
                   ! First apply a constant lift to induce the
+                  kernel = zero
+                  !if (dist2 < dist2_max) Kernel = prefactor * exp( - dist2 * invepsilon2)
+                  if (dist2 > (xlx*yly*zlz)) Kernel = prefactor * exp( - dist2 * invepsilon2)
```

Comment:
The original condition was : `if (dist<10.0*epsilon) then`

The new code is broken compared with the original.